service/controller/network.go

```diff
@@ -41,17 +38,30 @@ type Network struct {
 }
 
 func NewNetwork(config NetworkConfig) (*Network, error) {
-	if config.G8sClient == nil {
-		return nil, microerror.Maskf(invalidConfigError, "config.G8sClient must not be empty")
+	if config.K8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.K8sClient must not be empty")
 	}
 
 	var err error
 
+	var crdClient *k8scrdclient.CRDClient
```

Comment:
You can get the CRD Client from the client collection as well. Same for the scheme being used. Check the interface: https://github.com/giantswarm/k8sclient/blob/master/spec.go#L15-L25.